bin/compile

```diff
@@ -262,6 +272,8 @@ install_bins() {
     meta_set "build-step" "install-pnpm-using-corepack"
     if [[ "$package_manager" == pnpm* ]]; then
       package_manager="$package_manager"
+    elif [ -n "$pnpm_engine" ]; then
+      package_manager="pnpm@${pnpm_engine}"
     else
       warn "Currently pnpm installation is only supported through the 'packageManager' field in package.json"
```

Comment:
This error message seems to be untrue with the changes in this PR. Maybe it should be something like "The pnpm package manager was detected, but the pnpm version couldn't be determined. Please set 'packageManager' in package.json" or something.

Reply:
This would be replaced by https://github.com/heroku/heroku-buildpack-nodejs/pull/1226 which removes this generic warning and provides support for the default version.